solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -852,6 +815,68 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         return outputAmount.add(_reserveAmount.mul(compN).div(compD));
     }
 
+    /**
+      * @dev returns the amount of tokens required for liquidation
+      *
+      * @param _poolToken       pool token
+      * @param _reserveToken    reserve token
+      * @param _poolAmount      pool token amount when the liquidity was added
+      * @param _targetAmount    target token amount received in return
+      * @param _isBaseAmount    true if the required amount is of the based token
+      * @return amount of tokens required for liquidation
+    */
+    function liquidationAmount(
+        IDSToken _poolToken,
+        IERC20Token _reserveToken,
+        uint256 _poolAmount,
+        uint256 _targetAmount,
+        bool _isBaseAmount)
+        internal view returns (uint256)
+    {
+        // calculate the amount of pool tokens required for liquidation
+        // note that the amount is doubled since it's not possible to liquidate one reserve only
+        Fraction memory poolRate = poolTokenRate(_poolToken, _reserveToken);
+        uint256 poolAmount = _targetAmount.mul(poolRate.d).div(poolRate.n / 2);
+
+        // limit the amount of pool tokens by the amount that the system holds
+        uint256 systemBalance = store.systemBalance(_poolToken).add(_poolAmount);
```

Comment:
I understand why we add the pool amount to the system balance here but only because I'm familiar with the previous version of the code - it's really unclear in the context of the function itself - we take the amount that was initially added (based on the description) and we add it to the system balance and then limit the pool amount by that value? why would we limit it by a number that - in the context of the function - looks like is higher than the total number that the system holds?

Reply:
This nevertheless does the same thing that the previous code version did, with the intention of reusing code in favor of reducing code size. If there is no functional error in this new version, can you please point out what exactly it is that you want to fix, or better yet (assuming it's essentially cosmetic) - fix it yourself? Thanks.

Reply:
check the `removeLiquidity` function in the previous version, it doesn't add the pool amount to the system balance -

```
// limit the amount of pool tokens by the amount the system holds
uint256 systemBalance = store.systemBalance(liquidity.poolToken);
poolAmount = poolAmount > systemBalance ? systemBalance : poolAmount;
```

functional error as far as I can tell.

Reply:
It does in the other place, and I am passing `0` in this case.